### DISCUSSION TITLE
Problems running tests directly from Intellij when using Kotlin

### DISCUSSION THREAD

**Discussion by leeturner** [micronaut-projects/micronaut-core#7554] [Q&A]
Hi

I was wondering if anyone here is building their Micronaut projects using Intellij ?  I have just started playing around with Micronaut so I am sure I am missing something obvious but I don't seem to be able to get my tests to run directly from inside IntelliJ.

I thought it could be something specific to my project so to remove any doubt I have just pushed a [brand new project](https://github.com/leeturner/micronaut-intellij-kotlin-test-example) direct from Micronaut Launch to my github account. I haven’t changed any of the mvn settings or added any micronaut features or any code of my own. I run `mvn test` and it builds and runs the generated test fine. If I run a `mvn clean` and enable annotation processing in the Intellij settings I can’t run the test and get the error:

```
org.junit.jupiter.engine.execution.ConditionEvaluationException: Failed to evaluate condition [io.micronaut.test.extensions.junit5.MicronautJunit5Extension]: @MicronautTest used on test but no bean definition for the test present. This error indicates a misconfigured build or IDE. Please add the 'micronaut-inject-java' annotation processor to your test processor path (for Java this is the testAnnotationProcessor scope, for Kotlin kaptTest and for Groovy testCompile). See the documentation for reference: https://micronaut-projects.github.io/micronaut-test/latest/guide/
```

I followed the instructions here to enable annotation processing - https://docs.micronaut.io/latest/guide/index.html#ideSetup

If I run `man test` to compile the project and run the test, I can then successfully run the test from inside Intellij. I am using java 17 and kotlin:  

```
<jdk.version>17</jdk.version>
<release.version>17</release.version>
<micronaut.version>3.5.1</micronaut.version>
<micronaut.runtime>netty</micronaut.runtime>
<kotlinVersion>1.6.10</kotlinVersion>
```

I am sure I am missing something obvious but I don't know what it is.  Any help would be appreciated.

**Comment by leeturner** [comment]
I have just tried the same as above - downloaded a brand new project direct from Micronaut Launch and opened it up in Intellij.  I enabled annotation processing and ran the test and it all worked fine.  The only difference between these two projects was that this time I selected `java` as the language instead of `kotlin`

I wonder what it is about the kotlin project that means IntelliJ isn't processing the annotations correctly ?

**Comment by leeturner** [comment]
After more searching I have also found this - https://github.com/micronaut-projects/micronaut-core/issues/1936

Which lead me to this - https://youtrack.jetbrains.com/issue/KT-15040

Sorry for the noise.  I guess this is still an issue with the kotlin IntelliJ build system ?

**Reply by graemerocher** [reply]
Still an issue

**Reply by leeturner** [reply]
Amazing to think that the ticket has been open for 6 years. Looks like I'll switch back to Java for now. 

I did have some luck by making IntelliJ delegate to maven for build/run operations but it really slows down the flow a little.
